static/js/popovers.js

```diff
@@ -173,15 +173,23 @@ exports.toggle_actions_popover = function (element, id) {
         var editability = message_edit.get_editability(message);
         var use_edit_icon;
         var editability_menu_item;
-        if (editability === message_edit.editability_types.FULL) {
-            use_edit_icon = true;
-            editability_menu_item = i18n.t("Edit");
-        } else if (editability === message_edit.editability_types.TOPIC_ONLY) {
+        var cancel_editability_icon = false;
+        var row = current_msg_list.get_row(id);
+        if ($(row[0]).find('.message_content').css('display') === 'none') {
             use_edit_icon = false;
-            editability_menu_item = i18n.t("View Source / Edit Topic");
+            cancel_editability_icon = true;
+            editability_menu_item = i18n.t("Back to Message");
         } else {
```

Comment:
Can you make this an `else if` to avoid needing to add a layer of indentation to all the following code?